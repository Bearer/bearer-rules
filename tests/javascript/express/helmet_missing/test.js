const { createInvoker, getEnvironment } = require("../../../helper.js")
const { ruleId, ruleFile, testBase } = getEnvironment(__dirname)

describe(ruleId, () => {
  const invoke = createInvoker(ruleId, ruleFile, testBase)
  

  test("index", () => {
    const testCase = "index.ts"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  

  test("insecure", () => {
    const testCase = "insecure.js"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  

  test("secure", () => {
    const testCase = "secure.js"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  

  test("secure_part", () => {
    const testCase = "secure_part.js"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
})