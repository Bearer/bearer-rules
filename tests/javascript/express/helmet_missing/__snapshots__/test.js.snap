// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`javascript_express_helmet_missing index 1`] = `"{}"`;

exports[`javascript_express_helmet_missing insecure 1`] = `
"{
  "low": [
    {
      "cwe_ids": [
        "693"
      ],
      "id": "javascript_express_helmet_missing",
      "title": "Security misconfiguration detected (Helmet missing).",
      "description": "## Description\\n\\nHelmet can help protect your app from some well-known web vulnerabilities by setting HTTP headers appropriately.\\n\\n## Remediations\\n\\nâœ… Use Helmet middleware\\n\\n\`\`\`javascript\\nconst helmet = require(\\"helmet\\")\\napp.use(helmet())\\n\`\`\`\\n\\n## Resources\\n\\n- [Express Security Best Practices: Use Helmet](https://expressjs.com/en/advanced/best-practice-security.html#use-helmet)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/javascript_express_helmet_missing",
      "line_number": 4,
      "full_filename": "/tmp/bearer-scan/insecure.js",
      "filename": ".",
      "source": {
        "start": 4,
        "end": 4,
        "column": {
          "start": 11,
          "end": 15
        }
      },
      "sink": {
        "start": 4,
        "end": 4,
        "column": {
          "start": 11,
          "end": 15
        },
        "content": "ex()"
      },
      "parent_line_number": 4,
      "snippet": "ex()",
      "fingerprint": "124650418e34cb20fdd3bfe81d2a05e2_0",
      "old_fingerprint": "623670335300b582638d97b40e11dc87_0"
    }
  ]
}"
`;

exports[`javascript_express_helmet_missing secure 1`] = `"{}"`;

exports[`javascript_express_helmet_missing secure_part 1`] = `"{}"`;
